nvflare/private/fed/client/communicator.py

```diff
@@ -27,8 +27,8 @@
 from nvflare.apis.fl_constant import FLContextKey
 from nvflare.apis.fl_context import FLContext
 from nvflare.apis.fl_exception import FLCommunicationError
-from nvflare.apis.utils.local_logger import LocalLogger
-from nvflare.private.fed.utils.fed_utils import shareable_to_modeldata, make_context_data, make_shareeable_data
+from nvflare.app_common.widgets.log_streaming import LocalLogger
```

Comment:
We have a dependency violation here - private (which is part of the core) must not depend on app_common!
Looks like LocalLogger needs to be moved into the "core". We probably need to create a "local root logger" before firing ABOUT_TO_START_RUN (which will cause the LogSender to be added to the root logger).

Reply:
Thanks, this will be solved by PR #139.

Reply:
I will rebase after that is merged.